main.go

```diff
@@ -17,34 +18,58 @@ import (
 	"os/exec"
 	"regexp"
 	"strings"
+	"sync"
+	"time"
 )
 
+type output struct {
+	content string
+	mu      *sync.Mutex
+}
+
 // Cmd is typically constructed through the Command() call and provides state
 // to the execution engine.
 type Cmd struct {
 	cmd       *exec.Cmd
 	env       []string
 	exitError error
-	executed  bool
-	stdout    string
-	stderr    string
+	status    string
+	stdout    *output
+	stderr    *output
 	stdin     io.Reader
 }
 
 // ErrUninitializedCmd is returned when members are accessed before a run, that
 // can only be used after a command has been run.
 var ErrUninitializedCmd = errors.New("You need to run this command first")
-var pkgCmd = &Cmd{}
+// ErrCmdNotFinished is returned when members are accessed before or during a run,
+// that can only be used after a command has finished executing.
+var ErrCmdNotFinished = errors.New("Command is still executing")
+
+var pkgCmd = &Cmd{
+	status: "initialized",
+	stdout: &output{mu: &sync.Mutex{}},
+	stderr: &output{mu: &sync.Mutex{}},
+}
 
 // Command constructs a *Cmd. It is passed the command name and arguments.
 func Command(name string, arg ...string) *Cmd {
 	return &Cmd{
-		cmd: exec.Command(name, arg...),
+		cmd:    exec.Command(name, arg...),
+		status: "initialized",
+		stdout: &output{mu: &sync.Mutex{}},
+		stderr: &output{mu: &sync.Mutex{}},
 	}
 }
 
-func (c *Cmd) validate() {
-	if !c.executed {
+func (c *Cmd) validateIsDone() {
+	if c.status != "executed" {
+		log.Fatal(ErrCmdNotFinished)
```

Comment:
Based on my other comment about all the possible states, I would suggest to use "Finished" for the status instead of "executed".